src/test/java/org/opensearch/neuralsearch/sparse/algorithm/BatchClusteringTaskTests.java

```
@@ -0,0 +1,157 @@
+
+        // Verify task is created
+        assertNotNull("Task should be created successfully", task);
+
+        Field termsField = BatchClusteringTask.class.getDeclaredField("terms");
+        termsField.setAccessible(true);
+        @SuppressWarnings("unchecked")
+        List<BytesRef> taskTerms = (List<BytesRef>) termsField.get(task);
+
+        // Verify deep copy by checking actual content
+        assertEquals("First term should be 'term1'", "term1", taskTerms.get(0).utf8ToString());
+
+        // Modify original terms to verify deep copy
+        originalTerms.get(0).bytes[0] = (byte) 'X';
+
+        // Verify task's terms remain unchanged (proving deep copy worked)
+        assertEquals("Task's first term should still be 'term1'", "term1", taskTerms.get(0).utf8ToString());
+        assertNotEquals("Original term should now be different", "term1", originalTerms.get(0).utf8ToString());
+    }
+
+    public void testTaskCreationWithDifferentParameters() throws Exception {
+        // Test creating tasks with different parameter combinations
+        float[] summaryPruneRatios = { 0.1f, 0.5f, 0.9f };
+        float[] clusterRatios = { 0.0f, 0.3f, 1.0f };
+        int[] nPostingsValues = { 1, 10, 100 };
+
+        for (float summaryPruneRatio : summaryPruneRatios) {
+            for (float clusterRatio : clusterRatios) {
+                for (int nPostings : nPostingsValues) {
+                    BatchClusteringTask task = new BatchClusteringTask(terms, key, summaryPruneRatio, clusterRatio, nPostings, null, null);
+                    assertNotNull(
+                            "Task should be created with parameters: " + summaryPruneRatio + ", " + clusterRatio + ", " + nPostings,
+                            task
+                    );
+                }
+            }
+        }
+    }
+
+    public void testGetWithNullMergeState() throws Exception {
```

(it's not about this test) I think we could annotate member variable `mergeState` `NonNull`

---

if we expect it to be nonnull, we can make it fails early by constraining it to be nonnull, it's not about how test cases test it

---

OK. If later needs to modify BatchClusteringTask class, we can set it nonnull.